src/channel/mpmc.rs

```diff
@@ -771,7 +794,30 @@ mod if_alloc {
             where
                 T: Send,
             {
-                generic_channel::<parking_lot::RawMutex, T>(0)
+                generic_channel::<parking_lot::RawMutex, T, HeapRingBuf<T>>(0)
+            }
+
+            /// A [`GenericSender`] implementation backed by [`parking_lot`] whose
+            /// capacity grows dynamically up to the given limit.
```

Comment:
We might want to add some more documentation that differentiates it better from `HeapRingBuf` and which also gives recommendations on what the use when. But that can certainly be made in different CRs.